specs/BrowserProcessExited.md

```diff
@@ -0,0 +1,322 @@
+# Background
+We have heard asks for a WebView2 API to easily track the WebView2 Runtime's
+browser process exit. Manually waiting for the process to exit requires
+additional work on the host app, so we are proposing the `BrowserProcessExited`
+event. The `ProcessFailed` event already lets app developers handle unexpected
+browser process exits for a WebView, this new API lets you listen to both
+expected and unexpected termination of the processes associated to an
+environment from the `ICoreWebView2Environment3` interface so you can, e.g.,
+cleanup the user data folder when it's no longer in use. In this document we
+describe the new API. We'd appreciate your feedback.
+
+
+# Description
+The `BrowserProcessExited` event allows developers to subscribe event handlers
+to be run when the WebView2 Runtime processes (including the browser process)
+associated to a `CoreWebView2Environment` terminate. Key scenarios are cleanup
+of the user data folder used by the WebView2 Runtime, which is locked while the
+runtime's browser process is active, and moving to a new WebView2 Runtime
+version after a `NewBrowserVersionAvailable` event.
+
+This event is raised for both expected and unexpected browser process
+termination, after all resources, including the user data folder, used by the
+browser process (and related processes) have been released. The
+`ICoreWebView2BrowserProcessExitedEventArgs` interface lets app developers get
+the `BrowserProcessExitKind` so they can decide how to handle different exit
+kinds or bypass handling if an event handler for the `CoreWebView2`s
+`ProcessFailed` event (for `CoreWebView2ProcessFailedKind.BrowserProcessFailed`)
+is already registered. In case of a browser process crash, both
+`BrowserProcessExited` and `ProcessFailed` events are raised, but the order is
+not guaranteed.
```

Comment:
Discussion: the ordering isn't guaranteed because the events are coming from different sources out of our control. The intended scenarios for the two events are distinct. It is up to the app to coordinate the handlers so they don't try to perform reliability recovery during trying to update or remove user data folder.

Update docs to note this.

Reply:
Docs updated to note this.